src/pinnable_mapped_file.cpp

```diff
@@ -181,6 +222,81 @@ pinnable_mapped_file::pinnable_mapped_file(const std::filesystem::path& dir, boo
    }
 }
 
+void pinnable_mapped_file::setup_copy_on_write_mapping() {
+   // before we clear the Soft-Dirty bits for the whole process, make sure all writable,
+   // non-sharable chainbase dbs using mapped mode are flushed to disk
+   // ----------------------------------------------------------------------------------
+   for (auto pmm : _instance_tracker)
+      pmm->save_database_file(true);
```

Comment:
This mechanism doesn't look thread safe at all -- if someone ctors a chainbase on a thread it can cause corruption for any other chainbase in the process if they're busy being written to in a different thread.

I don't think we need to protect against that since leap never does that.

But since we still maintain this project a submodule for others to use, I wonder if we should somehow mention this limitation anywhere (in the header file or something?).. not sure

Reply:
Good point, I'll add a mention.

Reply:
done.